### PR TITLE
`sympa.pl --sync_include`: No errors were reported to caller

### LINKED ISSUE #907: Sync include from commandline oblivious of errors
Errors happening during synchronization of data sources don't bubble up to the caller.
```

/usr/local/sympa/bin/sympa.pl --sync_include=bart@lists.example.com
Members of list bart@lists.example.com have been successfully updated.

```
It doesn't matter whether there is an invalid `.incl` file or a SQL problem, the hapless user thinks everything is order.

### PR DESCRIPTION
**Pull request by ikedas**
`sympa.pl --sync_include`: No errors weren't reported to caller, even if one or more of data sources failed inclusion.

This PR depends on PR #905 .

This PR may fix #907 .

### CODE REVIEW

File: src/lib/Sympa/Request/Handler/include.pm [hunk 1]
```diff
@@ -280,7 +292,7 @@ sub _twist {
         $request, @result{qw(added deleted updated)}
     );
     $self->add_stash($request, 'notice', 'include_performed',
```
Comment (racke) [inline src/lib/Sympa/Request/Handler/include.pm]: This notice will be shown regardless of the outcome of the inclusion:

```
sympa@local-sympa:/var/lib/sympa/list_data/lists.example.com/test$ /usr/local/sympa/bin/sympa.pl --sync_include=test@lists.example.com
include [notice] Including subscribers from data source "subscribers.txt" into list test failed.
include [notice] Inclusion succeeded (0 added, 0 deleted, 0 updated).
```

**Comment by racke** [comment]
It would be also useful to add "skipped" to the result hash so the user is aware of bad data from the inclusion source.

File: src/lib/Sympa/Request/Handler/include.pm (hunk 1 above)
Comment (ikedas) [reply src/lib/Sympa/Request/Handler/include.pm]: Yes.  If multiple data sources are configured for a single list, some may succeed and some may fail.  `include_performed` report simply informs that inclusion task has been finised. What is more suitable text? "Inclusion finished"?

**Comment by ikedas** [comment]
> It would be also useful to add "skipped" to the result hash so the user is aware of bad data from the inclusion source.

`added`, `deleted` and `updated` are total numbers of records (users) successfully processed. I think number of `skipped` records can't be known.

**Comment by ikedas** [comment]
> It would be also useful to add "skipped" to the result hash so the user is aware of bad data from the inclusion source.

@racke, though I don't know this is what you suggested, I added numb er of "held" users: See cd429cb.

File: src/lib/Sympa/Request/Handler/include.pm (hunk 1 above)
Comment (ikedas) [reply src/lib/Sympa/Request/Handler/include.pm]: I separated `include_incomplete` and `include_performed`. See cd429cb.

File: default/mail_tt2/report.tt2 [hunk 2]
```diff
@@ -384,6 +384,9 @@
         [%|loc(report_param.name,report_param.listname)%]Including subscribers from data source "%1" into list %2 failed.[%END%]
     [%~ END %]
 
+[%~ ELSIF report_entry == 'include_incomplete' ~%]
+   [%|loc(report_param.result.added,report_param.result.held,report_param.result.updated)%]Inclusion partially or entirely failed (%1 added, %2 held, %3 updated).[%END%]
+
```
Comment (racke) [inline default/mail_tt2/report.tt2]: I would prefer to break this in two cases (partially / entirely).

We can determine whether a inclusion was partially or entirely failed, so I don't see a point in mixing them together. I also assume that in most cases we have a single datasource, so no need to confuse these users with partially failures.

Comment (ikedas) [reply default/mail_tt2/report.tt2]: Partial and entire cases were separated. See ee42f48

**Comment by racke** [comment]
There seems to be a logic error. So I have an inclusion from a file which doesn't exist:

```
Mar 20 08:42:11 local-sympa sympa[11984]: notice Sympa::Spindle::ProcessRequest::_twist() Processing Sympa::Request <action=include;context=test@lists.example.com;role=member>
Mar 20 08:42:11 local-sympa sympa[11984]: err main::#846 > Sympa::Spindle::spin#95 > Sympa::Request::Handler::include::_twist#185 > Sympa::Request::Handler::include::_update_users#343 > Sympa::DataSource::open#134 > Sympa::DataSource::File::_open#40 Unable to open file "/var/lib/sympa/list_data/lists.example.com/test/subscribers.txt": No such file or directory
Mar 20 08:42:11 local-sympa sympa[11984]: info Sympa::Request::Handler::include::_twist() Sympa::Request <action=include;context=test@lists.example.com;role=member>: Success, 0 added, 0 deleted, 0 updated
```

The error happens in "# II. Include new entries." but you do the accounting in " # III. Expire outdated entries.". 

I also think failed data sources should be excluded from further steps, thus remove them from them from _@dss_ before going to the expiration step.

**Comment by ikedas** [comment]
> The error happens in "# II. Include new entries." but you do the accounting in " # III. Expire outdated entries.".

Step III (expiration) can happen only once in one execution of include request, even if there are multiple data sources.  Because, if a user will no longer be included by updating of one data source, they can be included by another data source

**Comment by ikedas** [comment]
I re-pushed. @racke, could you please check the new patches? Thanks.

**Comment by racke** [comment]
Complete failure and success are looking good to me now. I will try to test partial failure as well.

**Comment by ikedas** [comment]
@racke, did you find anything signing problem? If you don't have complaint, could you please merge this pr?

**Pull request merged**